CloudFormation/test01_vpc.yml

```
@@ -0,0 +1,174 @@
+        - Key: Name
+          Value: !Sub ${NameBase}-subnet-public2-${AZc}
+
+
+  PrivateSubnet1:
+    Type: AWS::EC2::Subnet
+    Properties:
+      AvailabilityZone: !Ref AZa
+      VpcId: !Ref lecture10VPC
+      CidrBlock: 10.0.128.0/20
+      Tags:
+        - Key: Name
+          Value: !Sub ${NameBase}-subnet-private1-${AZa}
+
+
+  PrivateSubnet2:
+    Type: AWS::EC2::Subnet
+    Properties:
+      AvailabilityZone: !Ref AZc
+      VpcId: !Ref lecture10VPC
+      CidrBlock: 10.0.144.0/20
+      Tags:
+        - Key: Name
+          Value: !Sub ${NameBase}-subnet-private2-${AZc}
+# ------------------------------------------------------------#
+#  InternetGateway
+# ------------------------------------------------------------#
+  lecture10VPCIGW:
+    Type: AWS::EC2::InternetGateway
+    Properties:
+      Tags:
+        - Key: Name
+          Value:  !Sub ${NameBase}-igw
+  lecture10VPCAttachGateway:
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      VpcId: !Ref lecture10VPC
+      InternetGatewayId: !Ref lecture10VPCIGW
+
+# ------------------------------------------------------------#
+#  RouteTable
+# ------------------------------------------------------------#
+  PublicRouteTable:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref lecture10VPC
+      Tags:
+        - Key: Name
+          Value: !Sub ${NameBase}-rtb-public
+
+  PublicRoute:
+    Type: AWS::EC2::Route
+    Properties:
+      RouteTableId: !Ref PublicRouteTable
+      DestinationCidrBlock: 0.0.0.0/0
+      GatewayId: !Ref lecture10VPCIGW
+
+  PublicSubnet1Association:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: !Ref PublicSubnet1
+      RouteTableId: !Ref PublicRouteTable
+
+  PublicSubnet2Association:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: !Ref PublicSubnet2
+      RouteTableId: !Ref PublicRouteTable
+
+  PrivateRouteTable1:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref lecture10VPC
+      Tags:
+        - Key: Name
+          Value: !Sub ${NameBase}-rtb-private1-${AZa}
+  PrivateRouteTable2:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref lecture10VPC
+      Tags:
+        - Key: Name
+          Value: !Sub ${NameBase}-rtb-private1-${AZc}
+  PrivateSubnet1Associatio:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: !Ref PrivateSubnet1
+      RouteTableId: !Ref PrivateRouteTable1
+
+  PrivateSubnet2Associatio:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      SubnetId: !Ref PrivateSubnet2
+      RouteTableId: !Ref PrivateRouteTable2
+
+Outputs:
+  lecture10VPCID:
+    Description: information about VPCID
+    Value: !Ref lecture10VPC
+    Export: 
+      Name: !Sub ${AWS::StackName}-VPCID
+
+  PublicSubnet1:
+    Description: informarion about publicSubnet1
+    Value: !Ref PublicSubnet1
+    Export:
+      Name: !Sub ${AWS::StackName}-PublicSubnet1-a-ID
```

```suggestion
      Name: !Sub ${AWS::StackName}-PublicSubnet1-${AZa}-ID
```
ここも変数化しないと、Resourceセクションでの変更が反映されませんね！

---

「-a」、「-c」となっていた部分を変数化しました！